app/src/util/literalai_util.py

```diff
@@ -62,7 +71,19 @@ def get_all_entities[T](api_call: Callable[[LiteralClient, Any], PaginatedRespon
 
 
 def query_threads_between(start_date: datetime, end_date: datetime) -> list[Thread]:
-    return get_threads(filter_between(start_date, end_date))
+    threads = get_threads(filter_between(start_date, end_date))
+    save_entities(
+        threads, f"threads-{start_date.strftime('%Y-%m-%d')}-{end_date.strftime('%Y-%m-%d')}"
+    )
+    return threads
+
+
+def query_scores_between(start_date: datetime, end_date: datetime) -> list[Score]:
```

Comment:
There's an argument for removing this since it's not called but I used it to confirm that the `get_scores` API endpoint doesn't return any additional data beyond what is returned by the `get_threads` API endpoint. There's the opposing argument that this is an API client class that provides access to an external API, so we don't _have to_ call every available function. I can go either way.